### ISSUE TITLE
How to crash FontForge

### ISSUE THREAD

**Issue by SWWEFHoiiqwdjxqo** [fontforge/fontforge#4268]
FontForge has many ways to crash. These include:

• Triple click a bitmap glyph. This is a devastating crash, because FontForge will not even restore the bitmap glyphs upon importing!

• Copy a reference from one bitmap font to another. You now have a cross-font reference, except that when you save and reopen, FontForge crashes. It also crashes when the other font is closed and the glyph is attempted to be accessed.

**Comment by skef** [comment]
We haven't had many reports submitted on the bitmap font side of the code recently and I don't personally do anything with bitmap fonts. However, if you were to beef up this report a little with a specific test file and specific steps to reproduce (if only to overcome my ignorance) I'll take a look at the crashes and likely submit PRs to fix them.

**Comment by SWWEFHoiiqwdjxqo** [comment]
Cross-font references are probably an obscure PostScript feature that didn't quite get correctly implemented.

**Comment by skef** [comment]
The project is not very enthusiastic about chasing down every Fuzz-generated crash, but when there are crashes that arise in practice when attempting to use a program feature, even one that doesn't work very reliably, someone will often take a look and try to fix the crash.

**Comment by SWWEFHoiiqwdjxqo** [comment]
Do you know why triple-clicking a bitmap glyph not only crashes but also deletes the restore data for the bitmap glyphs?

**Comment by skef** [comment]
No, because I don't use the bitmap interface so I've never experienced the problem myself. 

I know it seems tedious but it actually takes a while to sort through an entirely new UI to reproduce things so I'm much more likely to look when the submitter includes a specific set of steps to reproduce.

**Comment by SWWEFHoiiqwdjxqo** [comment]
I can't test right now because FontForge proceeds to crash on Windows Vista.

**Comment by jtanx** [comment]
Try again with a newer os.

**Comment by SWWEFHoiiqwdjxqo** [comment]
Please unclose this. I have tested the March 14th version (2020) and the bugs are similar to what I remember encountering, crash still persists.

Here is a movie demonstrating the two crashes: https://www.youtube.com/watch?v=ISQa7xTQRVg

**Comment by skef** [comment]
I reopened the bug but it's going to be a *long* time before I try following that video to reproduce. You do, what, roughly 40 specific things before each crash?

**Comment by SWWEFHoiiqwdjxqo** [comment]
The first crash is in a bitmap font, click a glyph many times rapidly. One time it will lead to a crash.
The second crash is to copy a reference from one bitmap font to another, then close the font the reference was copied from.

**Comment by yg8ijvjvjv** [comment]
What bitmap font standard supports cross font references?

**Comment by Tynach** [comment]
> The first crash is in a bitmap font, click a glyph many times rapidly. One time it will lead to a crash. The second crash is to copy a reference from one bitmap font to another, then close the font the reference was copied from.

Wait, is it even valid to have one font reference a glyph from another font? I would *expect* that to give an error, but I don't know enough about the spec to say for sure.

The other problem, with rapid clicking... I can't reproduce. I use bitmap fonts a *lot* (as making one is the only reason I use FontForge at all) and have never run into this, at least on Linux. Even when I tried to reproduce it, I can't.

Maybe it has something to do with fonts that aren't saved, and are mostly or entirely empty?
